department-of-veterans-affairs/next-build
# VACMS-15668 cache layer

## Description
Closes #[15668](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15668).

Adds a cache layer (redis) to our drupalClient for production builds. This means `yarn build` and `yarn export` will both use the cache, `yarn dev` will not.

## Testing done
I ran `time yarn export` on the main branch and against this branch and compared the differences
local no cache: `yarn export  2371.92s user 36.34s system 294% cpu 13:37.13 total`
locally caching header + footer menus: `yarn export  753.89s user 24.78s system 578% cpu 2:14.64 total`

Nearly 11 minutes faster with just those 3 responses cached!


## Screenshots


## QA steps
`brew install redis` and then `redis-server` to start redis locally

in a separate terminal window, `yarn dev` and visit any page

in a separate terminal window, `redis-cli --scan --pattern '*'` should return nothing at this point. no cache utilized.

stop `yarn dev` and run `yarn export`. It should complete much faster than it has. When it completes, run `redis-cli --scan --pattern '*'` again and see:
```
"menu:footer-bottom-rail"
"menu:header-megamenu"
"menu:va-gov-footer"
```
You can also type `redis-cli` with no arguments to enter the redis interface itself. `KEYS *` will surface equivalent information (but will be much slower as we add more items to the cache)


```[tasklist]
```

## Is this PR blocked by another PR?
- Add the `DO NOT MERGE` label
- Add links to additional PRs here:
